bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,433 @@
+/**
+ * External dependencies
+ */
+
+import Octokit from '@octokit/rest';
+import fs from 'fs';
+
+import { fileURLToPath } from 'url';
+import nodePath, { dirname } from 'path';
+
+function getArg( argName ) {
+	const arg = process.argv.find( ( arg ) =>
+		arg.startsWith( `--${ argName }=` )
+	);
+	return arg ? arg.split( '=' )[ 1 ] : null;
+}
+
+const OWNER = 'wordpress';
+const REPO = 'gutenberg';
+
+const IGNORED_PATHS = [
+	'lib/experiments-page.php',
+	'packages/e2e-tests/plugins',
+	'packages/block-library', // this is handled automatically.
+];
+
+const DEBUG = !! getArg( 'debug' );
+
+const __filename = fileURLToPath( import.meta.url );
+const __dirname = dirname( __filename );
+
+const authToken = getArg( 'token' );
+
+async function main() {
+	if ( ! authToken ) {
+		console.error(
+			'Error. The --token argument is required. See: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token'
+		);
+		process.exit( 1 );
+	}
+
+	const sinceArg = getArg( 'since' );
+	let since;
+
+	if ( sinceArg ) {
+		if ( validateSince( sinceArg ) ) {
+			since = sinceArg;
+		} else {
+			console.error(
+				'Error: The --since argument cannot be more than 2 months from the current date.'
+			);
+			process.exit( 1 );
+		}
+	} else {
+		console.error(
+			`Error. The --since argument is required (e.g. YYYY-MM-DD). This should be the date of the previous release's final RC.`
+		);
+		process.exit( 1 );
+	}
+
+	console.log( 'Welcome to the PHP Sync Issue Generator!' );
+
+	if ( DEBUG ) {
+		console.log( 'DEBUG MODE' );
+	}
+
+	const octokit = new Octokit( {
+		auth: authToken,
+	} );
+
+	console.log( '--------------------------------' );
+	console.log( '• Running script...' );
+
+	// These should be paths where we expect to find PHP files that
+	// will require syncing to WordPress Core. This list should be
+	// extremely selective.
+	const paths = [ '/lib', '/phpunit' ];
+
+	console.log( `• Fetching all commits made to ${ REPO } since: ${ since }` );
```

Comment:
This is just merge commits to trunk, right?

Reply:
It uses https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits and the `path` arg to get the commits.

Reply:
Confirmed this defaults to `trunk`.